### PR TITLE
[SPARK-48659][SQL][TESTS] Unify v1 and v2 ALTER TABLE .. SET TBLPROPERTIES tests

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
- Move parser tests from `o.a.s.s.c.p.DDLParserSuite` and `o.a.s.s.e.c.DDLParserSuite` to `AlterTableSetTblPropertiesParserSuite`.
 - Porting and refactoring DS v1 tests from `DDLSuite` and `HiveDDLSuite` to `v1.AlterTableSetTblPropertiesBase` and to `v1.AlterTableSetTblPropertiesSuite`.
- Add a test for DSv2 `ALTER TABLE .. RENAME` to `v2.AlterTableSetTblPropertiesSuite`.

### Why are the changes needed?
- To improve test coverage.
- Align with other similar tests, eg: `AlterTableRename*`

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Add new UT & Update existed UT.
-  By running new test suites:
    ```
    $ build/sbt -Phive -Phive-thriftserver "test:testOnly *AlterTableSetTblPropertiesSuite"
    $ build/sbt -Phive -Phive-thriftserver "test:testOnly *AlterTableSetTblPropertiesParserSuite"
    ```

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala [hunk 1]
```diff
@@ -1075,19 +1075,11 @@ class DDLParserSuite extends AnalysisTest {
         ifExists = true))
   }
 
-  // ALTER TABLE table_name SET TBLPROPERTIES ('comment' = new_comment);
```
Comment (panbingkun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala:1078]: Only tests related to `SET` have been moved, and tests related to `UNSET` will doing in another PR

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetTblPropertiesParserSuite.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.apache.spark.sql.test.SharedSparkSession
+
+class AlterTableSetTblPropertiesParserSuite extends AnalysisTest with SharedSparkSession {
+
+  private def parseException(sqlText: String): SparkThrowable = {
+    intercept[ParseException](sql(sqlText).collect())
+  }
+
+  // ALTER TABLE table_name SET TBLPROPERTIES ('comment' = new_comment);
+  test("alter table: alter table properties") {
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetTblPropertiesParserSuite.scala:34]: Ported from ``

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetTblPropertiesParserSuite.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    val sql1_table = "ALTER TABLE table_name SET TBLPROPERTIES ('test' = 'test', " +
+      "'comment' = 'new_comment')"
+    comparePlans(
+      parsePlan(sql1_table),
+      SetTableProperties(
+        UnresolvedTable(Seq("table_name"), "ALTER TABLE ... SET TBLPROPERTIES", true),
+        Map("test" -> "test", "comment" -> "new_comment")))
+  }
+
+  test("alter table - property values must be set") {
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetTblPropertiesParserSuite.scala:44]: Ported from `sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala`

**Comment by panbingkun** [comment]
@cloud-fan Can you take a time to help review? Thank you very much.
Additionally, I have discovered an issue where we only have the `AlterNamespaceSetPropertiesExec` and not the `AlterNamespaceUnsetPropertiesExec`. Do we need to implement it?

**Comment by panbingkun** [comment]
cc @HyukjinKwon

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by cloud-fan** [comment]
Yea let's add AlterNamespaceUnsetPropertiesExec for DS v2

**Comment by panbingkun** [comment]
> Yea let's add AlterNamespaceUnsetPropertiesExec for DS v2

Thank you for your review! 
I will also do `Unify v1 and v2 ALTER TABLE .. UNSET TBLPROPERTIES tests` later.

**Pull request closed without merging**